src/pydicom/dataset.py

```diff
@@ -911,14 +895,63 @@ def __getattr__(self, name: str) -> Any:
         return object.__getattribute__(self, name)
 
     @property
-    def _character_set(self) -> list[str]:
+    def _character_set(self) -> str | list[str]:
         """The character set used to encode text values."""
         char_set = self.get(BaseTag(0x00080005), None)
         if not char_set:
             return self._parent_encoding
 
         return convert_encodings(char_set.value)
 
+    @property
+    def original_character_set(self) -> str | MutableSequence[str]:
+        """Return the original character set encoding for a dataset decoded
+        from a file or buffer.
+
+        Returns
+        -------
+        str | MutableSequence[str] | None
+            The original character set encoding of the dataset as given by
+            the (0008,0005) *Specific Character Set*, or `iso8859
+            <https://docs.python.org/3/library/codecs.html#standard-encodings>`_
+            if the dataset has been created from scratch.
+        """
+        return self._read_charset
+
+    @property
+    def read_encoding(self) -> str | MutableSequence[str]:
+        name = type(self).__name__
+        warnings.warn(
+            (
+                f"'{name}.read_encoding' will be removed in v4.0, use "
+                f"'{name}.original_character_set' instead"
+            ),
+            DeprecationWarning,
+        )
+
```

Comment:
Do we want to continue the [future behavior idea](https://pydicom.github.io/pydicom/stable/old/best_practices.html#future-proofing-your-code)? (I think it can be valuable to some people).  If so, then these should throw an error if 'future' is on.

Reply:
It's up to you, I'm not fussed either way. I might add it in stage 2 though, just so I can progress forwards on this one?

Reply:
Sure, stage 2 is fine